### PR TITLE
OutcomeExceptions: remove '__module__ = "builtins"' hack

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: src/_pytest/recwarn.py [hunk 1]
```diff
@@ -108,8 +108,7 @@ def warns(  # noqa: F811
         ...     warnings.warn("this is not here", UserWarning)
         Traceback (most recent call last):
           ...
-        Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
-
+        _pytest.outcomes.Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
```
Comment (blueyed) [inline src/_pytest/recwarn.py]: I think that's fine - and is in line what we get in other places with py3 compared to py2.

Comment (RonnyPfannschmidt) [reply src/_pytest/recwarn.py]: i do wonder if we want a follow-up to wrap the python trace-back formatting to ensure consistent display (with a way to opt out)

Comment (blueyed) [reply src/_pytest/recwarn.py]: I guess in this case it is more related to doctest / Python internals, not the traceback module (https://github.com/python/cpython/blob/24ddd9c2d6ab61cbce7e68d6de36d4df9bd2c3fb/Lib/traceback.py#L567-L568).

Comment (RonnyPfannschmidt) [reply src/_pytest/recwarn.py]: i do wonder if we should perhaps take a look at setting it to `pytest` instead of leaving it at the internal name

Comment (blueyed) [reply src/_pytest/recwarn.py]: Makes sense.  I guess we could set `__qualname__` then.
But it should be the importable name, i.e. `pytest.fail.Exception` then.
Should I do that?

Comment (RonnyPfannschmidt) [reply src/_pytest/recwarn.py]: -1 on that

the things we do expose on pytest, can have pytest, the  ones hidden stay hidden for now

Comment (blueyed) [reply src/_pytest/recwarn.py]: Ok, what do you suggest then?
Keep the hack as is?
Or remove it via this PR (unchanged)?

Comment (RonnyPfannschmidt) [reply src/_pytest/recwarn.py]: upon reading on the computer i believe i misunderstood

i was suggesting to use a importable name, i didn't take into account that currently we only expose them on the functions

File: doc/en/warnings.rst [hunk 2]
```diff
@@ -276,7 +276,7 @@ argument ``match`` to assert that the exception matches a text or regex::
     ...     warnings.warn("this is not here", UserWarning)
     Traceback (most recent call last):
       ...
-    Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
+    _pytest.outcomes.Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
```
Comment (asottile) [inline doc/en/warnings.rst]: maybe `__module__ = 'pytest'` -- I think we do this elsehwere (assuming this is an exported name)

Comment (blueyed) [reply doc/en/warnings.rst]: See above, it is not importable like that.

Comment (asottile) [reply doc/en/warnings.rst]: shall we make it so?

Comment (blueyed) [reply doc/en/warnings.rst]: See above, feel free - closing.

**Pull request closed without merging**